test/test_selectors.py

```diff
@@ -0,0 +1,684 @@
+        self.addCleanup(signal.signal, signal.SIGALRM, sigalrm_handler)
+
+        # Start the timer for the interrupt.
+        self.set_alarm(SHORT_SELECT)
+
+        with self.assertTakesTime(upper=LONG_SELECT):
+            self.assertEqual([(key, selectors.EVENT_READ)], s.select(LONG_SELECT))
+        self.assertEqual(rd.recv(1), b'x')
+
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_multiple_interrupts_with_event(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+        key = s.get_key(rd)
+
+        def second_alarm(*args):
+            wr.send(b'x')
+
+        def first_alarm(*args):
+            self.set_alarm(SHORT_SELECT)
+            signal.signal(signal.SIGALRM, second_alarm)
+
+        sigalrm_handler = signal.signal(signal.SIGALRM, first_alarm)
+        self.addCleanup(signal.signal, signal.SIGALRM, sigalrm_handler)
+        self.addCleanup(signal.alarm, 0)
+
+        # Start the timer for the interrupt.
+        self.set_alarm(SHORT_SELECT)
+
+        with self.assertTakesTime(lower=SHORT_SELECT * 2, upper=LONG_SELECT):
+            self.assertEqual([(key, selectors.EVENT_READ)], s.select(LONG_SELECT))
+        self.assertEqual(rd.recv(1), b'x')
+
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_selector_error(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+
+        def alarm_exception(*args):
+            err = OSError()
+            err.errno = 100
+            raise err
+
+        sigalrm_handler = signal.signal(signal.SIGALRM, alarm_exception)
+        self.addCleanup(signal.signal, signal.SIGALRM, sigalrm_handler)
+
+        # Start the timer for the interrupt.
+        self.set_alarm(SHORT_SELECT)
+
+        try:
+            s.select(LONG_SELECT)
+            self.fail("select() didn't raise SelectorError")
+        except selectors.SelectorError as e:
+            self.assertEqual(e.errno, 100)
+        except Exception:
+            self.fail("Raised incorrect exception")
+
+    # Test ensures that _syscall_wrapper properly raises the
+    # exception that is raised from an interrupt handler.
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_interrupt_exception(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+
+        class AlarmInterrupt(Exception):
+            pass
+
+        def alarm_exception(*args):
+            raise AlarmInterrupt()
+
+        sigalrm_handler = signal.signal(signal.SIGALRM, alarm_exception)
+        self.addCleanup(signal.signal, signal.SIGALRM, sigalrm_handler)
+
+        # Start the timer for the interrupt.
+        self.set_alarm(SHORT_SELECT)
+
+        with self.assertTakesTime(upper=LONG_SELECT):
```

Comment:
No you are totally correct this should only take `SHORT_SELECT` time.  Good catch.